Jenkinsfile

```diff
@@ -71,31 +74,49 @@ pipeline {
               return env.BUILD_FILEBEAT != "false"
             }
           }
-          stages {
-            stage('Filebeat oss'){
-              steps {
-                withBeatsEnv(){
-                  makeTarget("Filebeat oss Linux", "-C filebeat testsuite")
-                }
-              }
+          steps {
+            makeTarget("Filebeat oss Linux", "-C filebeat testsuite")
+          }
+        }
+        stage('Filebeat x-pack'){
+          agent { label 'ubuntu && immutable' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            expression {
+              return env.BUILD_FILEBEAT != "false"
```

Comment:
The `env.BUILD_FILEBEAT` value doesn't account for changes in the `^x-pack/filebeat` directories, does it? We probably want to run the x-pack build/test when there's any change in the OSS side or x-pack dir (since x-pack depends on oss).

Reply:
https://github.com/elastic/beats/pull/15570/files#r372308026